### PR TITLE
chore(typography): convert enzyme tests to RTL

### PR DESCRIPTION
**Pull request by nuria1110**
### Proposed behaviour

Convert enzyme tests to RTL. 

### Current behaviour

Tests are in enzyme.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

File: src/components/typography/typography.test.tsx [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+    margin: "-1px",
+    overflow: "hidden",
+    padding: "0",
+    position: "absolute",
+    width: "1px",
+    whiteSpace: "nowrap",
+  });
+});
+
+test("should override 'display' property when passed", () => {
```
Comment (tomdavies73) [inline src/components/typography/typography.test.tsx:280]: **question**: was there any particular reason why this was included from the override tests, coverage?

File: src/components/typography/typography.test.tsx [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+  render(<Typography truncate>Test</Typography>);
+
+  expect(screen.getByText("Test")).toHaveStyle({
+    whiteSpace: "nowrap",
+    textOverflow: "ellipsis",
+    overflow: "hidden",
+  });
+});
+
+test("should render with visuallyHidden styles when screenReaderOnly is true", () => {
```
Comment (tomdavies73) [inline src/components/typography/typography.test.tsx:265]: **nit-pick** (non-blocking): wrap prop name in apostrophes to match other tests

File: src/components/typography/typography.test.tsx (hunk 1 above)
Comment (nuria1110) [reply src/components/typography/typography.test.tsx:280]: Yeah this one was for coverage

File: src/components/typography/typography.stories.tsx [hunk 3]
```diff
@@ -71,8 +71,8 @@ export const VariantsStory: Story = () => (
       </Typography>
     </Typography>
     <Typography variant="small" display="block" mb={1}>
-      The small variant renders a small elemevant, which is used for small
-      print. The small variant also has a smaller appearance.
+      The small variant renders a small element, which is used for small print.
```
Comment (DipperTheDan) [inline src/components/typography/typography.stories.tsx:74]: **praise(non-blocking):** Well spotted.

**Review by DipperTheDan — COMMENTED** [review]
Nothing more to add from me. Just address the comments left by @tomdavies73 and I'll happily approve.

**Review by DipperTheDan — APPROVED** [review]

**Review by tomdavies73 — APPROVED** [review]

**Review by stephenogorman — APPROVED** [review]

**Pull request merged**